### PR TITLE
Adding BloodHound Enterprise Solution

### PR DESCRIPTION
**Pull request by gwhite-so**
Required items, please complete
   
   Change(s):
   - Adding BloodHound Enterprise Solution

   Reason for Change(s):
   - Creating new Sentinel Solution

   Testing Completed:
   - Yes

### CODE REVIEW

**Comment by gwhite-so** [comment]
> @gwhite-so please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

**Comment by v-prasadboke** [comment]
Hello @gwhite-so looking into this

**Comment by v-prasadboke** [comment]
Hello @gwhite-so please create a package by going through this txt file ' https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/README.md '

**Comment by v-prasadboke** [comment]
Hello @gwhite-so please include input file in the solution folder as well,
Under folder name Data.
' Solutions/BloodHound Enterprise/Data/Solution_BloodHoundEnterprise '

**Comment by gwhite-so** [comment]
Hi @v-prasadboke, 

It appears the solution is failing on Kqlvalidations and detectionTemplateSchemaValidation. Do the files below need to be updated in my PR for the validation tests to pass?

https://github.com/Azure/Azure-Sentinel/tree/master/.script/tests/KqlvalidationsTests/CustomTables
https://github.com/Azure/Azure-Sentinel/blob/master/.script/tests/detectionTemplateSchemaValidation/ValidConnectorIds.json

**Comment by v-prasadboke** [comment]
Hello @gwhite-so let me go through this

**Comment by v-prasadboke** [comment]
Ok @gwhite-so looking into this

File: Sample Data/Custom/bloodhoundEnterprise_CL.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        "data_type": "paths"
+    },
+    {
+        "domain_id": "S-1-5-21-3130019616-2776909439-2417379567",
+        "domain_name": "TESTLAB.LOCAL",
+        "path_id": "T0FindingId",
+        "path_title": "Example Path Title",
+        "group": null,
+        "principal": null,
+        "non_tier_zero_principal": "NON TIER ZERO@TESTLAB.LOCAL",
```
Comment (v-prasadboke) [inline Sample Data/Custom/bloodhoundEnterprise_CL.json:49]: Hello @gwhite-so can you change the value of  "non_tier_zero_principal" & "tier_zero_principal".
The ' @ ' is causing validation failure, can you please fix it

Comment (gwhite-so) [reply Sample Data/Custom/bloodhoundEnterprise_CL.json]: Hi @v-prasadboke, I'm not sure what the Azure pipeline validation is failing on. The error message is unclear about what the issue is.

Comment (gwhite-so) [reply Sample Data/Custom/bloodhoundEnterprise_CL.json]: Hi @v-prasadboke, I removed the '@' sign from the sample data, I'm not sure what portion of the data connector the Azure pipeline is now failing on.

**Comment by v-prasadboke** [comment]
Hello @gwhite-so please update the branch from master

**Review by v-prasadboke — COMMENTED** [review]
Please add workbook metadata in "C:\GitHub\Azure-Sentinel\Tools\Create-Azure-Sentinel-Solution\V2\WorkbookMetadata\WorkbooksMetadata.json" file

File: Solutions/BloodHound Enterprise/Parsers/BloodHoundEnterprise.txt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+let BloodHound_view  = view () { 
+  bloodhoundEnterprise_CL
+  | extend
+      EventVendor="SpecterOps",
+      EventProduct="BloodHound Enterprise",
+      domain_sid=coalesce(column_ifexists('domain_sid_s', ''), column_ifexists('domain_sid_g', '')),
```
Comment (v-prasadboke) [inline Solutions/BloodHound Enterprise/Parsers/BloodHoundEnterprise.txt:6]: Hello @gwhite-so please replace domain_sid_g with,domain_sid_s

**Comment by v-prasadboke** [comment]
Hello @gwhite-so the branch was not updated from master i have updated it now

**Comment by v-prasadboke** [comment]
Hello @gwhite-so dataconnector schema in not, Ill raise a PR to update the data connector schema in the individual folder.

**Comment by v-prasadboke** [comment]
I have raised a PR for the Data connector schema #8562
Thanks

**Comment by v-prasadboke** [comment]
Hello @gwhite-so after making these changes, please repackage the solution using V3 tool. 
Please go through this readme file for instructions on using V3 tool

**Comment by gwhite-so** [comment]
Hi @v-prasadboke, I've made all the requested changes and repackaged the solution using the V3 tool. The new commit appears to be passing all of the validations.

**Comment by v-prasadboke** [comment]
Hello @gwhite-so thanks for the updates. Will be inspecting this and you can expect an update till 21 Jul 2023

**Comment by gwhite-so** [comment]
Hi @v-prasadboke 

I've made all of the requested minor changes to the solution.

Thanks,
Garrett

**Comment by v-prasadboke** [comment]
Hello @gwhite-so, Thank you for the Contribution.
I just committed a minor change related to readme file where a pipeline was missing.

**Review by v-prasadboke — APPROVED** [review]
All looks good

**Review by v-prasadboke — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**